pkg/controllers/dynakube/extension/eec/statefulset.go

```diff
@@ -273,11 +272,33 @@ func buildContainerVolumeMounts(dk *dynakube.DynaKube) []corev1.VolumeMount {
 		})
 	}
 
+	if dk.Spec.Templates.ExtensionExecutionController.TlsRefName != "" {
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      consts.ExtensionsCustomTlsCertificate,
+			MountPath: baseSecretsPath + "/eec",
+			ReadOnly:  true,
+		})
+	} else {
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      httpsCertVolumeName,
+			MountPath: httpsCertMountPath,
+			ReadOnly:  true,
+		})
+	}
+
 	return volumeMounts
 }
 
 func setVolumes(dk *dynakube.DynaKube) func(o *appsv1.StatefulSet) {
 	return func(o *appsv1.StatefulSet) {
+		tlsSecretName := extensionsControllerTlsSecretName
+		tlsCertVolumeName := httpsCertVolumeName
+
+		if dk.Spec.Templates.ExtensionExecutionController.TlsRefName != "" {
+			tlsSecretName = dk.Spec.Templates.ExtensionExecutionController.TlsRefName
+			tlsCertVolumeName = consts.ExtensionsCustomTlsCertificate
```

Comment:
why two different Volume names are used?

Reply:
Same as above, it shouldn't make a difference who creates the cert.

Reply:
fixed in bc76d466